### DISCUSSION TITLE
[TWITCH] Extract/remove ads from finished `output.ts` file

### DISCUSSION THREAD

**Discussion by marcopixel** [streamlink/streamlink#5087] [General]
I've tried to record a livestream on Twitch and even though i had `--twitch-disable-ads` enabled it saved the ads into the video stream into the output.ts file. But there seems to be some sort of metadata inside the file because it will show 30sec timestamps when playing via VLC - so my question is if anyone knows a way to remove/extract those ads from the video?

Haven't found anything with FFMPEG or MP4BOX, but i might be looking at the wrong thing though. 😊

**Comment by bastimeyer** [comment]
> even though i had `--twitch-disable-ads` enabled it saved the ads into the video stream into the output.ts file

No, it does not. You either didn't have it set, or you were running an older version of Streamlink. The recent version(s) of Streamlink properly filter out ad segments from Twitch streams. Without a full debug log, your claim is moot.

HLS streams on Twitch also don't consist of (fragmented) mp4 segments. They are MPEG-2 transport streams.

Segments just get concatenated, so if there are any stream discontinuities, like when filtering out ads, and especially when not filtering out ads, then the output simply contains broken timestamps and is thus incoherent. That's how it is, and it can't be fixed. More about that here: #4535

**Comment by marcopixel** [comment]
Thank you for your explanation, it was a error on my side (the script i'm using was incorrectly setting the wrong option via the python API) and that's why the recording included the ads into the video stream. 😄
